frog/imports/ui/Repository.jsx

```diff
@@ -1,9 +1,230 @@
-import React, { Component } from 'react';
- 
-const Repository = () => { 
-  return(
-      <p>You are on the Repository app</p>
-  )
+import React, { Component, PropTypes } from 'react';
+import { Activities } from '../api/activities.js';
+import { createContainer } from 'meteor/react-meteor-data';
+import ReactDOM from 'react-dom';
+
+export default class Repository extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      nameFilterText:"",
+      //The elements in the following lists are the attributes of the activities
+      //we want to filter
+      plane: [],
+      type: [],
```

Comment:
I used the lists to make it easier to add a new filter (by just adding one line of code), but now that I think about, the way you suggested makes the code clearer about what kind of filters are available. I'll make the necessary changes.